core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -3696,14 +3696,16 @@ class KafkaApis(val requestChannel: RequestChannel,
       requestContext,
       DESCRIBE,
       TOPIC,
-      groupDescribeOffsetsRequest.topics.asScala
+      groupDescribeOffsetsRequest.topics.valuesList.asScala
     )(_.topicName)
 
+    val topicCollection = new DescribeShareGroupOffsetsRequestData.DescribeShareGroupOffsetsRequestTopicCollection()
+    authorizedTopics.foreach(t => topicCollection.add(t.duplicate))
```

Comment:
I don't see why we now need duplication where previously we did not. Could you explain? Thanks.